modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/client/Iroha2Client.kt

```diff
@@ -139,6 +139,25 @@ open class Iroha2Client(
         }
     }
 
+    private var lastRequestedPeerIdx: Int? = null
+
+    // Round-robin load balancing
+    protected fun getTelemetryUrl() = getUrls().second
+
+    // Round-robin load balancing
+    protected fun getPeerUrl() = getUrls().first
+
+    private fun getUrls() = when (lastRequestedPeerIdx) {
+        null -> urls.first().also { lastRequestedPeerIdx = 0 }
+        else -> {
+            lastRequestedPeerIdx = when (lastRequestedPeerIdx) {
+                urls.size - 1 -> 0
+                else -> lastRequestedPeerIdx!! + 1
+            }
+            urls[lastRequestedPeerIdx!!]
```

Comment:
how about healthcheck before sending a request? can be an optional field of  the strategy I mentioned